src/main/java/ay2122s1_cs2103t_w16_2/btbb/model/order/UniqueOrderList.java

```diff
@@ -74,6 +81,50 @@ public void setOrders(List<Order> orders) {
         internalList.setAll(orders);
     }
 
+    /**
+     * Returns the top ten clients with the most orders.
+     *
+     * @return Top ten clients with the most orders.
+     */
+    public List<Map.Entry<Pair<GenericString, Phone>, Integer>> getTopTenClients() {
+        HashMap<Pair<GenericString, Phone>, Integer> clientOrderCountKeyValuePair =
+                getClientOrderCountKeyValuePair();
+
+        List<Map.Entry<Pair<GenericString, Phone>, Integer>> topTenClients =
+                getTopTenClientsFromHashMap(clientOrderCountKeyValuePair);
+
+        return topTenClients;
+    }
+
+    private HashMap<Pair<GenericString, Phone>, Integer> getClientOrderCountKeyValuePair() {
+        HashMap<Pair<GenericString, Phone>, Integer> clientOrderCountKeyValuePair = new HashMap<>();
+
+        internalList.stream().forEach(order -> {
+            GenericString clientName = order.getClientName();
+            Phone clientPhone = order.getClientPhone();
+            Pair<GenericString, Phone> clientNameAndPhone = new Pair<>(clientName, clientPhone);
+            if (clientOrderCountKeyValuePair.containsKey(clientNameAndPhone)) {
+                int orderCount = clientOrderCountKeyValuePair.get(clientNameAndPhone);
+                clientOrderCountKeyValuePair.put(clientNameAndPhone, ++orderCount);
+            } else {
+                clientOrderCountKeyValuePair.put(clientNameAndPhone, 1);
+            }
+        });
+
+        return clientOrderCountKeyValuePair;
+    }
+
+    private List<Map.Entry<Pair<GenericString, Phone>, Integer>> getTopTenClientsFromHashMap(
+            HashMap<Pair<GenericString, Phone>, Integer> hashMap) {
+        List<Map.Entry<Pair<GenericString, Phone>, Integer>> clientOrderCountPairs =
+                hashMap.entrySet()
+                        .stream()
+                        .sorted(Collections.reverseOrder(Map.Entry.comparingByValue()))
+                        .collect(Collectors.toList());
+
+        return clientOrderCountPairs.size() <= 10 ? clientOrderCountPairs : clientOrderCountPairs.subList(0, 10);
```

Comment:
We probably need to put a disclaimer that ties are arbitrarily broken. For example if you have 11 clients with the same number of orders, we can only pick a random 10.
```suggestion
                        .sorted(Collections.reverseOrder(Map.Entry.comparingByValue()))
                        .limit(10)
                        .collect(Collectors.toList());
                        
                        return clientOrderCountPairs;
```